### PR TITLE
feat: add webpack v4 support

### LINKED ISSUE #119: [FR]: support webpack 4
### What is the current behavior?

Worker crashes with webpack 4, not testing with webpack 4.

### Describe the feature

Webpack 4 is still widely used and probably worth supporting.

WIP showing some potential changes: https://github.com/aspect-build/rules_webpack/pull/124

### PR DESCRIPTION
**Pull request by jbedard**
Fix #119

### CODE REVIEW

File: webpack/private/webpack_worker.js [hunk 1]
```diff
@@ -46,21 +46,28 @@ class WebpackWorker extends WebpackCLI {
     if (err && this.reject) {
       this.console.error(err)
       this.reject(err)
+    } else if (stats.hasErrors() && this.reject) {
```
Comment (jbedard) [inline webpack/private/webpack_worker.js:49]: Lets merge this in https://github.com/aspect-build/rules_webpack/pull/130 first

File: webpack/private/webpack.config.js.tmpl [hunk 2]
```diff
@@ -1,22 +1,31 @@
...
-  },
-  output: {
+  const optimization = {
+    moduleIds: v4 ? 'natural' : 'deterministic',
+    chunkIds: v4 ? 'natural' : 'deterministic',
+  }
+
+  // Webpack4: passing any 'output' overrides the standard output config in entries
+  const output = v4 ? {
+    path: __dirname,
```
Comment (jbedard) [inline webpack/private/webpack.config.js.tmpl:20]: Without this it outputs to `dist/` by default, the webpack5 `uniqueName` must override the default `dist/`? 🤷

Comment (thesayyn) [reply webpack/private/webpack.config.js.tmpl:20]: what about `--output` flag we specify? doesn't that override this?

**Review by thesayyn — APPROVED** [review]

File: webpack/private/webpack.config.js.tmpl (hunk 2 above)
Comment (jbedard) [reply webpack/private/webpack.config.js.tmpl:20]: You would think so, but no it doesn't seem like it :(

**Pull request merged**